app/pages/1_📊_Explore.py

```diff
@@ -15,53 +17,119 @@
+    
+    # EXPONENTIAL transformation
+    y_exp = np.log(y)
+    model_exp = sm.OLS(y_exp, sm.add_constant(X)).fit()
+    y_pred_exp = np.exp(model_exp.predict(sm.add_constant(X)))
+    
+    # POLYNOMIAL transformation
+    poly = PolynomialFeatures(degree=3)
+    X_poly = poly.fit_transform(X)
+    model_poly = LinearRegression().fit(X_poly, y)
+    y_pred_poly = model_poly.predict(X_poly)
+    
+    # POWER LAW transformation
+    X_power = np.log(X)
+    y_power = np.log(y)
+    model_power = sm.OLS(y_power, sm.add_constant(X_power)).fit()
+    y_pred_power = np.exp(model_power.predict(sm.add_constant(X_power)))
+    
+    return X_log, y_pred_log, y_exp, y_pred_exp, y_pred_poly, X_power, y_power, y_pred_power
+
+def plot_transformations(df, x_col, y_col):
+    df = preprocess_data(df, x_col, y_col)
+    
+    try:
+        X_log, y_pred_log, y_exp, y_pred_exp, y_pred_poly, X_power, y_power, y_pred_power = apply_transformations(df, x_col, y_col)
         
-        This dataset contains the GDP per capita of USA from 1960-2023."""
-    )
-
-    df = pd.read_csv(f"datasets/{selected_dataset}")
-    st.write(df.head())
-    st.line_chart(df, x="Year", y="GDP_per_capita")
-
-if selected_dataset == "eco_fuel_consumption.csv":
-    st.write(
-        """
-        ## 
-        This synthetic dataset contains the EcoFuel consumption of a county's population."""
-    )
-
-    df = pd.read_csv(f"datasets/{selected_dataset}")
-    st.write(df.head())
-    st.line_chart(df, x="population", y="ecofuel_consumption")
-
-if selected_dataset == "reactions.csv":
-    st.write(
-        """
-        ## 
-        This synthetic dataset contains the Reactions when a concentration is added to an experiment"""
-    )
+        fig = make_subplots(rows=2, cols=2, subplot_titles=("LOG", "EXPONENTIAL", "POLYNOMIAL", "POWER LAW"))
+        
+        # LOG
+        fig.add_trace(go.Scatter(x=np.log(df[x_col]), y=df[y_col], mode='markers', name='Data'), row=1, col=1)
+        fig.add_trace(go.Scatter(x=np.log(df[x_col]), y=y_pred_log, mode='lines', name='LOG Fit'), row=1, col=1)
+        fig.update_xaxes(title_text=f"log({x_col})", row=1, col=1)
+        fig.update_yaxes(title_text=y_col, row=1, col=1)
+        
+        # EXPONENTIAL
+        fig.add_trace(go.Scatter(x=df[x_col], y=np.log(df[y_col]), mode='markers', name='Data'), row=1, col=2)
+        fig.add_trace(go.Scatter(x=df[x_col], y=np.log(y_pred_exp), mode='lines', name='EXPONENTIAL Fit'), row=1, col=2)
+        fig.update_xaxes(title_text=x_col, row=1, col=2)
+        fig.update_yaxes(title_text=f"log({y_col})", row=1, col=2)
+        
+        # POLYNOMIAL
+        fig.add_trace(go.Scatter(x=df[x_col], y=df[y_col], mode='markers', name='Data'), row=2, col=1)
+        fig.add_trace(go.Scatter(x=df[x_col], y=y_pred_poly, mode='lines', name='POLYNOMIAL Fit'), row=2, col=1)
+        fig.update_xaxes(title_text=x_col, row=2, col=1)
+        fig.update_yaxes(title_text=y_col, row=2, col=1)
```

Comment:
@nicktbrr Something is funny with this when the Infant Mortality dataset is used.

Reply:
I looked into it, but it might just be that for that dataset specifically X and y are on different scales. A possible fix might be to scale the data but that would kind of defeat the purpose. We could maybe add a note at the bottom?